cpp/cmake/Modules/JitifyPreprocessKernels.cmake

```diff
@@ -0,0 +1,73 @@
+#=============================================================================
+# Copyright (c) 2018-2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+
+cmake_minimum_required(VERSION 3.18)
+
+file(MAKE_DIRECTORY "${CUDF_GENERATED_INCLUDE_DIR}/include/jit_preprocessed_files")
+
+# Create `jitify_preprocess` executable
+project(jitify_preprocess VERSION 2.0 LANGUAGES CXX CUDA)
+add_executable(jitify_preprocess "${JITIFY_INCLUDE_DIR}/jitify2_preprocess.cpp")
+
+target_link_libraries(jitify_preprocess CUDA::cudart ${CMAKE_DL_LIBS})
+
+set(JIT_PREPROCESSED_FILES)
+
+function(jit_preprocess_files)
+    cmake_parse_arguments(ARG
+                          ""
+                          "SOURCE_DIRECTORY"
+                          "FILES"
+                          ${ARGN}
+                          )
+
+    foreach(ARG_FILE ${ARG_FILES})
+        set(ARG_OUTPUT ${CUDF_GENERATED_INCLUDE_DIR}/include/jit_preprocessed_files/${ARG_FILE}.jit)
+        set(JIT_PREPROCESSED_FILES "${ARG_OUTPUT};${JIT_PREPROCESSED_FILES}")
+        add_custom_command(WORKING_DIRECTORY ${ARG_SOURCE_DIRECTORY}
+                           DEPENDS jitify_preprocess
+                           OUTPUT ${ARG_OUTPUT}
+                           VERBATIM
+                           COMMAND ${CUDF_BINARY_DIR}/jitify_preprocess ${ARG_FILE}
+                                    -o ${CUDF_GENERATED_INCLUDE_DIR}/include/jit_preprocessed_files
+                                    -v
+                                    -i
+                                    -m
+                                    -std=c++14
+                                    -remove-unused-globals
+                                    -D__CUDACC_RTC__
+                                    -I${CUDF_SOURCE_DIR}/include
+                                    -I${CUDF_SOURCE_DIR}/src
+                                    # -I${LIBCXX_INCLUDE_DIR}
+                                    -I${LIBCUDACXX_INCLUDE_DIR}
+                                    -I${CUDAToolkit_INCLUDE_DIR}
```

Comment:
This should be `CUDAToolkit_INCLUDE_DIRS` as that is the public facing variable `CUDAToolkit_INCLUDE_DIR` is an implementation detail and can be removed in a future version